test/celeritas/optical/Generator.test.cc

```diff
@@ -168,6 +169,46 @@ TEST_F(LArSphereGeneratorTest, primary_generator)
     EXPECT_EQ(65536, gen.num_generated);
 }
 
+TEST_F(LArSphereGeneratorTest, direct_generator)
+{
+    // Create direct generator action
+    std::vector<optical::TrackInitializer> inits(
+        128,
+        optical::TrackInitializer{units::MevEnergy{1e-5},
+                                  Real3{0, 0, 0},
+                                  Real3{1, 0, 0},
+                                  Real3{0, 1, 0},
+                                  0,
+                                  ImplVolumeId{0}});
+    auto generate = optical::DirectGeneratorAction::make_and_insert(
+        *this->core(), *this->optical_params());
+
+    this->build_state<MemSpace::host>(32);
+    this->build_transporter();
```

Comment:
The transporter should be built before the state now so the aux data is correctly constructed:
```suggestion
    this->build_transporter();
    this->build_state<MemSpace::host>(32);
```